### PR TITLE
Add a note to setVariables

### PR DESCRIPTION
**Pull request by unirey**
Important note to `setVariables` if you're working with `first` and `last`. It's very easy to get stuck with this one if you're new to Relay. It's logical to someone who knows Relay well but not as much to beginner.

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by wincent** [comment]
Thanks @unirey. Are you able to CLA? If not, let me know so that I can close this out. Thanks!

**Comment by JenniferWang** [comment]
@facebook-github-bot import

File: docs/APIReference-Container.md [hunk 1]
```diff
@@ -383,9 +383,11 @@ module.exports = Relay.createContainer(Feed, {
 });
 ```
 
-> Note
+> Notes
 >
 > `setVariables` does not immediately mutate `variables`, but creates a  pending state transition. `variables` will continue returning the previous values until `this.props` has been populated with data that fulfills the new variable values.
+>
+> If `setVariables` is used to change `first` and `hasNextPage` is `false` (also `last` and `hasPreviousPage`), Relay does not send a request to the server because there isn't any more data as far as Relay is concerned.
```
Comment (josephsavona) [inline docs/APIReference-Container.md]: > because there isn't any more data as far as Relay is concerned.

It's not really Relay that is making this choice; it is the schema. The contract for `hasNextPage` is that it should be true if there are more edges to fetch, and false if there are no more edges. So it might be more accurate to rephrase this as "because the server has indicated there are no more edges that can be fetched."

**Comment by wincent** [comment]
Going to close this one due to inactivity. If you get to signing the CLA, @unirey, let us know and we'll re-open.
